frontend/src/app/(vendor)/partner/claim/page.tsx

```diff
@@ -0,0 +1,244 @@
+import { verifyVendorMagicLink } from "@/features/profile/common/api/magicLink";
+import { signUpAndClaimVendor } from "@/features/profile/manage/hooks/claimVendor";
+import { useNotification } from "@/contexts/NotificationContext";
+import { TOKEN_PARAM } from "@/lib/constants";
+import Visibility from "@mui/icons-material/Visibility";
+import VisibilityOff from "@mui/icons-material/VisibilityOff";
+
+function VendorClaimPageContent() {
+  const { addNotification } = useNotification();
+  const searchParams = useSearchParams();
+  const router = useRouter();
+  const supabase = createClient();
+
+  const token = searchParams?.get(TOKEN_PARAM) || "";
+
+  const [isLoading, setIsLoading] = useState(true);
+  const [isClaiming, setIsClaiming] = useState(false);
+  const [vendorInfo, setVendorInfo] = useState<{ name: string; email: string } | null>(null);
+  const [errorMessage, setErrorMessage] = useState("");
+
+  // Form fields
+  const [password, setPassword] = useState("");
+  const [confirmPassword, setConfirmPassword] = useState("");
+  const [showPassword, setShowPassword] = useState(false);
+
+  // Initialize page: verify token → load vendor info
+  useEffect(() => {
+    const init = async () => {
+      if (!token) {
+        setErrorMessage("Invalid claim link.");
+        setIsLoading(false);
+        return;
+      }
+
+      try {
+        const verification = await verifyVendorMagicLink("", "", token);
+
+        if (!verification.success) {
+          setErrorMessage("Invalid or expired claim link.");
+        } else {
+          setVendorInfo({
+            name: verification.vendorBusinessName || "Your Vendor",
+            email: verification.vendorEmail || "",
+          });
+        }
+      } catch {
+        setErrorMessage("Failed to verify claim link.");
+      }
+
+      setIsLoading(false);
+    };
+
+    init();
+  }, [token]);
+
+  const handleClaim = async (e: React.FormEvent) => {
+    e.preventDefault();
+
+    if (!password || !confirmPassword) {
+      setErrorMessage("Please fill out all fields.");
+      return;
+    }
+
+    if (password !== confirmPassword) {
+      setErrorMessage("Passwords do not match.");
+      return;
+    }
+
+    if (!vendorInfo?.email) {
+      setErrorMessage("Missing vendor email.");
+      return;
+    }
+
+    setIsClaiming(true);
+    setErrorMessage("");
+
+    try {
+      const result = await signUpAndClaimVendor(vendorInfo.email, token, password);
+
+      if (!result.success) {
+        setErrorMessage(result.error || "Failed to claim vendor.");
+        setIsClaiming(false);
+        return;
+      }
+
+      // Sign in automatically
+      const { error: signInError } = await supabase.auth.signInWithPassword({
+        email: vendorInfo.email,
+        password,
+      });
+
+      if (signInError) {
+        setErrorMessage("Account created, but sign-in failed. Please sign in manually.");
+        setIsClaiming(false);
+        return;
```

Comment:
should we redirect to the login page?

Reply:
I think the user should decide. we could add a login button?